utility/tests/test_repoutil.py

```diff
@@ -0,0 +1,260 @@
+		self.repo1.logger = logger
+		self.repo2.logger = logger
+		self.repo3.logger = logger
+		self.repo4.logger = logger
+
+		self.assertEqual(self.repo1.logger, logger)
+		self.assertEqual(self.repo2.logger, logger)
+		self.assertEqual(self.repo3.logger, logger)
+
+		self.myname = 'Stanislav Chlebec'
+		self.myemail = 'stanislav.chlebec@pantheon.tech'
+
+		self.myname5 = 'Sergej Testerko'
+		self.myemail5 = 'sergejtesterko@yandex.com'
+
+	def tearDown(self):
+		self.repo1.remove()
+		self.repo2.remove()
+		self.repo3.remove()
+		self.repo4.remove()
+		#self.repo5.remove()
+
+	def test_remove(self):
+
+		self.assertEqual(self.repo1.clone(self.myname, self.myemail), None)  
+		self.assertTrue(os.path.exists(self.repo1.localdir))
+		dir_be_removed = self.repo1.localdir
+
+		self.assertEqual(self.repo1.remove(), None)  
+		self.assertEqual(self.repo1.localdir, None)
+		self.assertEqual(self.repo1.repo, None)
+		self.assertFalse(os.path.exists(dir_be_removed))
+		
+		# try to remove already removed repo
+		self.assertEqual(self.repo1.remove(), None)  
+
+	def test_clone(self):
+
+		self.assertRaises(TypeError, self.repo1.clone, 1)
+
+		self.assertEqual(self.repo1.clone(self.myname, self.myemail), None)  
+		localdir = self.repo1.localdir
+		self.assertTrue(os.path.exists(localdir))
+		self.assertIsNotNone(self.repo1.repo)
+		with self.repo1.repo.config_reader() as config:
+			self.assertEqual(config.get_value('user','email'), self.myemail)
+			self.assertEqual(config.get_value('user','name'), self.myname)
+		#print('Repo1 cloned into: ' + self.repo1.localdir)
+
+		self.assertEqual(self.repo2.clone(self.myname, self.myemail), None)  
+		localdir = self.repo2.localdir
+		self.assertTrue(os.path.exists(localdir))
+		self.assertIsNotNone(self.repo2.repo)
+		with self.repo2.repo.config_reader() as config:
+			self.assertEqual(config.get_value('user','email'), self.myemail)
+			self.assertEqual(config.get_value('user','name'), self.myname)
+		#print('Repo2 cloned into: ' + self.repo2.localdir)
+
+		self.assertEqual(self.repo3.clone(self.myname, self.myemail), None)  
+		localdir = self.repo3.localdir
+		self.assertTrue(os.path.exists(localdir))
+		self.assertIsNotNone(self.repo3.repo)
+		with self.repo3.repo.config_reader() as config:
+			self.assertEqual(config.get_value('user','email'), self.myemail)
+			self.assertEqual(config.get_value('user','name'), self.myname)
+		#print('Repo3 cloned into: ' + self.repo3.localdir)
+
+		with self.assertRaises(GitCommandError):
+			self.repo4.clone(self.myname, self.myemail)
+
+
+	def test_get_repo_dir(self):
+
+		self.assertEqual(self.repo1.clone(self.myname, self.myemail), None)  
+		self.assertEqual(self.repo2.clone(self.myname, self.myemail), None)  
+		self.assertEqual(self.repo3.clone(self.myname, self.myemail), None)  
+
+		repodir1 = self.repo1.get_repo_dir()
+		repodir2 = self.repo2.get_repo_dir()
+		repodir3 = self.repo3.get_repo_dir()
+
+		msg = 'git clone would strip the suffix .git while cloning the repo'
```

Comment:
This actually depends on what you have as a url. so line 41
`		repourl1 = 'https://github.com/stanislav-chlebec/docs'`
if you would use
`		repourl1 = 'https://github.com/stanislav-chlebec/docs.git'`
then it would not strip the suffix.

The 165 is failing for me because you are using backend.git

Reply:
Well what I tried to indicate is that "git clone" would create folder without ".git" extension in both cases (1. URI without .git 2 URI wit .git)
THe name of tested function - get_repo_dir is somewhat misleading ...
But I see that it is not used in the same way as mentioned "git clone " command generates its folder for holding repo content
I will delete the code expecting that ".git" is stripped